pytential/symbolic/elasticity.py

```diff
@@ -0,0 +1,614 @@
+    def apply(self, density_vec_sym, dir_vec_sym, qbx_forced_limit,
+            extra_deriv_dirs=()):
+        """Symbolic expressions for integrating Stresslet kernel.
+
+        Returns an object array of symbolic expressions for the vector
+        resulting from integrating the dyadic Stresslet kernel with
+        variable *density_vec_sym* and source direction vectors *dir_vec_sym*.
+
+        :arg density_vec_sym: a symbolic vector variable for the density vector.
+        :arg dir_vec_sym: a symbolic vector variable for the direction vector.
+        :arg qbx_forced_limit: the *qbx_forced_limit* argument to be passed on
+            to :class:`~pytential.symbolic.primitives.IntG`.
+        :arg extra_deriv_dirs: adds target derivatives to all the integral
+            objects with the given derivative axis.
+        """
+        raise NotImplementedError
+
+    def apply_derivative(self, deriv_dir, density_vec_sym, dir_vec_sym,
+            qbx_forced_limit):
+        """Symbolic derivative of velocity from Stokeslet.
+
+        Returns an object array of symbolic expressions for the vector
+        resulting from integrating the *deriv_dir* target derivative of the
+        dyadic Stokeslet kernel with variable *density_vec_sym*.
+
+        :arg deriv_dir: integer denoting the axis direction for the derivative.
+        :arg density_vec_sym: a symbolic vector variable for the density vector.
+        :arg dir_vec_sym: a symbolic vector variable for the normal direction.
+        :arg qbx_forced_limit: the *qbx_forced_limit* argument to be passed on
+            to :class:`~pytential.symbolic.primitives.IntG`.
+        """
+        return self.apply(density_vec_sym, dir_vec_sym, qbx_forced_limit,
+                (deriv_dir,))
+
+# }}}
+
+
+# {{{ Naive and Biharmonic impl
+
+def _create_int_g(knl, deriv_dirs, density, **kwargs):
+    for deriv_dir in deriv_dirs:
+        knl = AxisTargetDerivative(deriv_dir, knl)
+
+    kernel_arg_names = set(karg.loopy_arg.name
+            for karg in (knl.get_args() + knl.get_source_args()))
+
+    # When the kernel is Laplace, mu and nu are not kernel arguments
+    # Also when nu==0.5, it's not a kernel argument to StokesletKernel
+    for var_name in ["mu", "nu"]:
+        if var_name not in kernel_arg_names:
+            kwargs.pop(var_name)
+
+    res = sym.int_g_vec(knl, density, **kwargs)
+    return res
+
+
+class _ElasticityWrapperNaiveOrBiharmonic:
+    def __init__(self, dim, mu_sym, nu_sym, base_kernel):
+        self.dim = dim
+        self.mu = mu_sym
+        self.nu = nu_sym
+
+        if not (dim == 3 or dim == 2):
+            raise ValueError("unsupported dimension given to ElasticityWrapper")
+
+        self.base_kernel = base_kernel
+
+        self.kernel_dict = {}
+        # The two cases of nu=0.5 and nu!=0.5 differ significantly and
+        # ElasticityKernel needs to know if nu=0.5 or not at creation time
+        poisson_ratio = "nu" if nu_sym != 0.5 else 0.5
```

Comment:
Fixed in 7f850899012603a3fb5a27ae7ef4d55f26e56a1a

Reply:
I'm still not sure this works in all cases. The scenario I had in mind was
* Calling `ElasticityWrapper(dim=3, mu_sym=sym.var("my_mu"), nu_sym=sym.var("my_nu"))` (where as far as I can tell those names get thrown away)
* In the `__init__` it calls `ElasticityKernel` from `sumpy` which has its own names for `mu` / `viscosity_mu` and `nu` / `poisson_ratio`
* Then, on `bind(...)(actx, my_mu=3, my_nu=0.1)` it will crash because it doesn't know the names

Does that make sense? Does it actually work? (Haven't tried it myself :(

Reply:
Those names are not thrown away. They are passed to the constructor of `IntG`.
Your example will not crash. I changed `test/test_stokes.py` to use `mu2` and `nu2` in this PR and it works.

Reply:
Ah, I see! It's all good then! Thank you for taking the time to clear that up! :rocket: